### PR TITLE
fix: update jwt.Claims

### PR DESCRIPTION
**Pull request by JeyJeyGao**
jwt.Claims is used as jws's payload which is required to be JSON compatible object.

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: signature/jws/envelope.go [hunk 1]
```diff
@@ -57,8 +58,16 @@ func (e *envelope) Sign(req *signature.SignRequest) ([]byte, error) {
 		return nil, err
 	}
 
+	// parse payload as jwt.MapClaims
```
Comment (patrickzheng200) [inline signature/jws/envelope.go:61]: Let's add a Reference URL to jwt.MapClaims in the comment.

File: signature/jws/envelope.go [hunk 2]
```diff
@@ -57,8 +58,16 @@ func (e *envelope) Sign(req *signature.SignRequest) ([]byte, error) {
...
+	var payload jwt.MapClaims
+	err = json.Unmarshal(req.Payload.Content, &payload)
```
Comment (binbin-li) [inline signature/jws/envelope.go:63]: nit: we can combine line 63 and 64

File: signature/jws/envelope_test.go [hunk 3]
```diff
@@ -563,7 +563,7 @@ func TestPayload(t *testing.T) {
 		checkNoError(t, err)
 
 		_, err = newEnv.Payload()
-		checkErrorEqual(t, "illegal base64 data at input byte 476", err.Error())
+		checkErrorEqual(t, "payload error: illegal base64 data at input byte 288", err.Error())
```
Comment (chloeyin) [inline signature/jws/envelope_test.go:566]: I found this 288 strange... Can we do a fuzz check. For example, checking the error message contains the keyword

File: signature/jws/envelope.go (hunk 1 above)
Comment (JeyJeyGao) [reply signature/jws/envelope.go:61]: Added.

File: signature/jws/envelope.go (hunk 2 above)
Comment (JeyJeyGao) [reply signature/jws/envelope.go]: Good idea.

File: signature/jws/envelope_test.go (hunk 3 above)
Comment (JeyJeyGao) [reply signature/jws/envelope_test.go]: Updated.
```
func checkErrorEqual(t *testing.T, want, got string) {
	if !strings.Contains(got, want) {
		t.Fatalf("want: %v got: %v\n", want, got)
	}
}
```

**Review by patrickzheng200 — APPROVED** [review]
lgtm

**Pull request merged**